File: Swift/ExampleTests/common/PatchTitleTyped.swift
public struct PatchTitleTyped: Equatable, CustomStringConvertible {
    public let title: String

    public init(title: String) {
        self.title = title
    }

    public init(from s: PatchTitleString) {
        self.title = s.title
    }

    public func toStringStruct() -> PatchTitleString {
        return PatchTitleString(
            title: String(describing: title)
        )
    }

    public static func toStringList(_ list: [PatchTitleTyped]) -> [PatchTitleString] {
        return list.map { $0.toStringStruct() }
    }

    public static func fromStringList(_ list: [PatchTitleString]) -> [PatchTitleTyped] {
        return list.map { PatchTitleTyped(from: $0) }
    }

    public func toJSONValue() -> [String: Any] {
        return [
            "title": title,
        ]
    }

    public func toJSON() throws -> String {
        return try Json.write(toJSONValue())
    }

    public init(fromJSONValue m: [String: Any]) throws {
        self.title = try Json.asString(Json.require(m, "title"), "title")
    }

    public init(fromJSON text: String) throws {
        try self.init(fromJSONValue: Json.parseObject(text))
    }

    public static func toJSONList(_ list: [PatchTitleTyped]) throws -> String {
        return try Json.write(list.map { $0.toJSONValue() })
    }

    public static func fromJSONList(_ text: String) throws -> [PatchTitleTyped] {
        return try Json.parseArray(text).map {
            try PatchTitleTyped(fromJSONValue: Json.asObject($0, "PatchTitleTyped"))
        }
    }

    public var description: String {
        return "title=\(title)"
    }
}

File: Rust/ExampleTests/common/patchtitle_typed.rs
#![allow(dead_code, unused_imports, unused_variables)]

use super::json;
use super::patchtitle_string::PatchTitleString;

#[derive(Debug, Clone, Default, PartialEq)]
pub struct PatchTitleTyped {
    pub title: String,
}

impl PatchTitleTyped {
    pub fn from_str_struct(s: &PatchTitleString) -> Self {
        Self {
            title: s.title.clone(),
        }
    }

    pub fn to_str_struct(&self) -> PatchTitleString {
        PatchTitleString {
            title: self.title.clone(),
        }
    }

    pub fn to_string_list(list: &[PatchTitleTyped]) -> Vec<PatchTitleString> {
        list.iter().map(|t| t.to_str_struct()).collect()
    }

    pub fn from_string_list(list: &[PatchTitleString]) -> Vec<PatchTitleTyped> {
        list.iter().map(PatchTitleTyped::from_str_struct).collect()
    }

    pub fn to_json_value(&self) -> json::Value {
        json::Value::Object(vec![
            ("title".to_string(), json::Value::Str(self.title.clone())),
        ])
    }

    pub fn to_json(&self) -> String {
        json::write(&self.to_json_value())
    }

    pub fn from_json_value(v: &json::Value) -> json::JsonResult<Self> {
        Ok(Self {
            title: json::as_string(json::require(v, "title")?, "title")?,
        })
    }

    pub fn from_json(text: &str) -> json::JsonResult<Self> {
        Self::from_json_value(&json::parse(text)?)
    }

    pub fn to_json_list(list: &[PatchTitleTyped]) -> String {
        json::write(&json::Value::Array(
            list.iter().map(|item| item.to_json_value()).collect(),
        ))
    }

    pub fn from_json_list(text: &str) -> json::JsonResult<Vec<PatchTitleTyped>> {
        let root = json::parse(text)?;
        let items = json::as_array(&root, "PatchTitleTyped")?;
        let mut result = Vec::with_capacity(items.len());
        for e in items {
            result.push(Self::from_json_value(e)?);
        }
        Ok(result)
    }
}
